tests/testthat/test-shinytest2-landing_popup.R

```diff
@@ -0,0 +1,184 @@
+testthat::test_that("e2e: teal app with landing_popup_module initializes with no errors", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(
+        title = "Welcome",
+        content = tags$b("A welcome message!", style = "color: red;")
+      ),
+      example_module()
+    )
+  )
+  app$view()
+
+  app$wait_for_idle(timeout = default_idle_timeout)
+  app$expect_no_shiny_error()
+  app$stop()
+})
+
+test_that("e2e: app with default landing_popup_module creates modal containing a button", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_equal(
+    app$get_html(".btn-default") %>%
+      grep("Accept", x = ., value = TRUE) %>%
+      rvest::read_html() %>%
+      rvest::html_text(),
+    "Accept"
+  )
+
+  app$stop()
+})
+
+test_that("e2e: when default landing_popup_module is closed, it shows the underlying teal app", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  app$click(selector = ".btn-default:nth-child(1)")
```

Comment:
How about clicking on the button that has a `data-dismiss` attribute?

```suggestion
  app$click(selector = "#shiny-modal-wrapper button[data-dismiss='modal']")
```

Or the first one, but inside the wrapper

```suggestion
  app$click(selector = "#shiny-modal-wrapper button:nth-child(1)")
```

Reply:
I always prefer selecting using data attributes over other ways, especially over getting the nth element. It will easily break once a new element is inserted/removed.